src/FunctionApps/PITest_FunctionApp/IngestFunction/__init__.py

```diff
@@ -0,0 +1,96 @@
+import logging
+import pysftp
+import sys
+import azure.functions as func
+from .settings import settings
+from azure.storage.blob import BlobServiceClient
+import uuid
+from io import BytesIO
+from pathlib import Path
+
+
+def print_tree(sftp: pysftp.Connection, path: str):
+    file_names = []
+    dir_names = []
+    other_names = []
+
+    def store_files(fname: str):
+        file_names.append(fname)
+
+    def store_dirs(dirname: str):
+        dir_names.append(dirname)
+
+    def store_other(name: str):
+        other_names.append(name)
+
+    sftp.walktree(path, store_files, store_dirs, store_other, recurse=True)
+    logging.info(
+        f"File names: {file_names}\nDirectory names: {dir_names}\nOther names: {other_names}"
+    )
+
+
+def upload_blob(local_file_name: str, data: BytesIO):
+    if not settings.connection_string:
+        exit(f"Connection string not set")
+    blob_service_client = BlobServiceClient.from_connection_string(
+        settings.connection_string
+    )
+    # Create a unique name for the container
+    container_name = str(uuid.uuid4())
+
+    # Create the container
+    logging.debug(f"Creating container {container_name}")
+
+    container_client = blob_service_client.create_container(container_name)
+    logging.debug("\nListing existing blobs...")
+
+    # List the blobs in the container
+    blob_list = container_client.list_blobs()
+    for blob in blob_list:
+        logging.debug("\t" + blob.name)
+
+    blob_client = blob_service_client.get_blob_client(
+        container=container_name, blob=local_file_name
+    )
+
+    logging.info(
+        "\nUploading passed-in blob to Azure Storage as blob:\n\t" + local_file_name
+    )
+    blob_client.upload_blob(data)
+
+
+def test_decryption(sftp: pysftp.Connection):
+    target_file_path = "/eICR/TEST_FILE.TXT"
+    target_file_name = Path(target_file_path).name
+    file_object = BytesIO()
+    sftp.getfo(target_file_path, file_object)
+    file_object.seek(0)
+    logging.debug(
+        f"Uploading {target_file_name} at path {target_file_path} Azure Storage"
+    )
+    upload_blob(target_file_name, file_object)
+
+
+def main(req: func.HttpRequest) -> func.HttpResponse:

```

Comment:
I am actually having a hard time tracking how this gets triggered to completion.
1) This function gets called and it creates the sftp connection
2) Print tree gets called...
3) I don't see where `test_decryption` is called which would then call `upload_blob` 

This is likely just me needing an explanation which I greatly appreciate.

Reply:
Ah, thanks for catching this - I need to remove the `test_decryption` method, as this was just meant for that single file; the solution really should attempt to run `get_r` against the dir tree to copy everything over . I had this change on a separate branch but neglected to remove this here

Reply:
Still, a high-level comment on the `main` function here would be nice

Reply:
So the action items I'm seeing here are:
1. delete the `test_decryption` function in its entirety.
2. Make a call _somewhere_ to `get_r` to recursively copy the directory structure from remote to local.

Does this also imply that we're dumping `upload_blob`, or is that something we'll call after we've copied the directories over?